Added Volume Weighted Moving Average

brhurley opened the pull request:
Fixes #.

Changes proposed in this pull request:
- Adding VWMAIndicator to the indicators.averages package 

- [X] added an entry _with related ticket number(s)_ to the unreleased section of `CHANGELOG.md`

brhurley commented:
Ready for review

brhurley commented:
@nimo23 When you get time can you review this?

TheCookieLab commented on ta4j-core/src/test/java/org/ta4j/core/indicators/averages/VWMAIndicatorTest.java:
```diff
@@ -0,0 +1,62 @@
+    @Test
+    public void vwmaIndicatorTest() {
+
+        MockIndicator mock = CsvTestUtils.getCsvFile(VWMAIndicatorTest.class, "VWMA.csv", numFactory);
```
Please flesh out these unit tests further. You can use the CsvTestUtils however they should be in addition to (not a replacement for) atomic test cases including edge conditions. 

When testing using mock data, there should be at minimum one unit test that demonstrates the calculation using simple numbers so that other users can easily reason about the indicator's logic at a glance (as opposed to digesting an entire CSVs worth context).

Please see `RecentSwingLowIndicatorTest `or `KalmanFilterIndicatorTest` as examples of how I expect indicator unit tests to be structured.

TheCookieLab reviewed (changes requested):
The PR's opening "post" should be filled out correctly. The template exists for a reason and all contributors are expected to do or provide reason why it's not applicable.

brhurley commented:
> The PR's opening "post" should be filled out correctly. The template exists for a reason and all contributors are expected to do or provide reason why it's not applicable.

I added a line in the PR but please update your template on what you expect.   I couldn't find a CHANGE.md so I updated the CHANGELOG.md file.   Also while I don't see any other examples of ticket numbers in the CHANGELOG.md file I put the PR reference number on my line.

I'll work on adding a unit test like you requested next.

TheCookieLab commented:
Gotcha, thanks,

On Sat, Dec 14, 2024 at 4:16 PM Brian Hurley ***@***.***>
wrote:

> The PR's opening "post" should be filled out correctly. The template
> exists for a reason and all contributors are expected to do or provide
> reason why it's not applicable.
>
> I added a line in the PR but please update your template on what you
> expect. I couldn't find a CHANGE.md so I updated the CHANGELOG.md file.
> Also while I don't see any other examples of ticket numbers in the
> CHANGELOG.md file I put the PR reference number on my line.
>
> I'll work on adding a unit test like you requested next.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ta4j/ta4j/pull/1238#issuecomment-2543347001>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABQJ6FN5ESYHKMFCRUVWJ5D2FSN3TAVCNFSM6AAAAABTPAL5W2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNBTGM2DOMBQGE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

brhurley commented:
@TheCookieLab updated as requested.
